File: scripts/generate-issues.py
#!/usr/bin/env python3
"""
GitHub Issue Generator für NdcApp Roadmap (Python Version)

Einfachere Alternative zum Node.js Script für Benutzer die Python bevorzugen.
"""

import os
import re
import json
from pathlib import Path
from typing import List, Dict, Any
from datetime import datetime


class RoadmapIssueGenerator:
    def __init__(self, roadmap_path: str = "../ROADMAP.md", output_dir: str = "./python-generated-issues"):
        self.roadmap_path = Path(roadmap_path)
        self.output_dir = Path(output_dir)
        self.issues = []
        
    def generate_issues(self):
        """Hauptfunktion zur Generierung aller Issues"""
        print("🚀 Starte GitHub Issue Generierung (Python)...")
        
        # Roadmap laden und parsen
        roadmap_content = self._load_roadmap()
        self.issues = self._parse_roadmap(roadmap_content)
        
        # Ausgabe-Verzeichnis erstellen
        self.output_dir.mkdir(exist_ok=True, parents=True)
        
        # Issues generieren
        self._generate_issue_files()
        self._generate_summary()
        self._generate_github_cli_script()
        
        print(f"✅ {len(self.issues)} Issues erfolgreich generiert!")
        print(f"📁 Ausgabeverzeichnis: {self.output_dir}")
        
    def _load_roadmap(self) -> str:
        """Lädt die Roadmap-Datei"""
        if not self.roadmap_path.exists():
            raise FileNotFoundError(f"Roadmap-Datei nicht gefunden: {self.roadmap_path}")
        
        with open(self.roadmap_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        print("📖 Roadmap-Datei geladen")
        return content
        
    def _parse_roadmap(self, content: str) -> List[Dict[str, Any]]:
        """Parst die Roadmap und extrahiert Tasks"""
        lines = content.split('\n')
        issues = []
        current_section = ""
        current_subsection = ""
        
        for i, line in enumerate(lines):
            line = line.strip()
            
            # Section-Header erkennen
            if line.startswith('## '):
                current_section = re.sub(r'[🎯🚀🏗️🔧📊🚦📞💡]', '', line.replace('## ', '')).strip()
                continue
                
            if line.startswith('### '):
                current_subsection = re.sub(r'[0-9.]', '', line.replace('### ', '')).strip()
                continue
                
            # Tasks erkennen (- [ ] Format)
            if line.startswith('- [ ] ') and self._should_process_section(current_section):
                task_text = line.replace('- [ ] ', '').replace('**', '').strip()
                
                if len(task_text) >= 10:  # Mindestlänge für valide Tasks
                    issue = {
                        'title': self._clean_title(task_text),
                        'body': self._generate_body(task_text, current_section, current_subsection),
                        'labels': self._generate_labels(current_section),
                        'priority': self._determine_priority(current_section),
                        'section': current_section,
                        'subsection': current_subsection
                    }
                    issues.append(issue)
        
        print(f"📋 {len(issues)} Tasks aus Roadmap extrahiert")
        return issues
        
    def _should_process_section(self, section: str) -> bool:
        """Prüft ob eine Section verarbeitet werden soll"""
        process_sections = [
            'Kurzzeitige Ziele',
            'Mittelfristige Ziele',
            'Langfristige Ziele', 
            'Technische Schulden',
            'Nächste Schritte'
        ]
        return any(s in section for s in process_sections)
        
    def _clean_title(self, title: str) -> str:
        """Bereinigt und kürzt den Titel"""
        title = re.sub(r'\*\*(.*?)\*\*', r'\1', title)  # Bold formatting entfernen
        if len(title) > 80:
            title = title[:77] + '...'
        return title
        
    def _generate_body(self, task_text: str, section: str, subsection: str) -> str:
        """Generiert den Issue-Body"""
        body = f"## Beschreibung\n\n{task_text}\n\n"
        
        if subsection:
            body += f"**Bereich:** {subsection}\n"
        body += f"**Roadmap-Kategorie:** {section}\n\n"
        
        body += "## Akzeptanzkriterien\n\n"
        body += "- [ ] Feature/Verbesserung ist implementiert\n"
        body += "- [ ] Tests sind vorhanden (falls anwendbar)\n"
        body += "- [ ] Dokumentation ist aktualisiert\n"
        body += "- [ ] Code Review ist durchgeführt\n\n"
        
        body += "## Zusätzliche Informationen\n\n"
        body += "Diese Issue wurde automatisch aus der [Roadmap](../ROADMAP.md) generiert.\n\n"
        body += "Siehe auch: [Vollständige Roadmap](../ROADMAP.md)"
        
        return body
        
    def _generate_labels(self, section: str) -> List[str]:
        """Generiert Labels basierend auf Section"""
        labels = ['roadmap']
        
        section_lower = section.lower()
        if 'code' in section_lower or 'qualität' in section_lower:
            labels.append('code-quality')
        elif 'dokumentation' in section_lower:
            labels.append('documentation')
        elif 'deployment' in section_lower:
            labels.append('deployment')
        else:
            labels.append('enhancement')
            
        # Priorität hinzufügen
        priority = self._determine_priority(section)
        labels.append(f'priority-{priority}')
        
        return labels
        
    def _determine_priority(self, section: str) -> str:
        """Bestimmt Priorität basierend auf Section"""
        if 'Kurzzeitige Ziele' in section:
            return 'high'
        elif 'Mittelfristige Ziele' in section or 'Technische Schulden' in section:
            return 'medium'
        elif 'Langfristige Ziele' in section:
            return 'low'
        return 'medium'
        
    def _generate_issue_files(self):
        """Generiert Issue-Dateien"""
        for i, issue in enumerate(self.issues):
            filename = f"issue-{i+1:03d}-{self._slugify(issue['title'])}.md"
            filepath = self.output_dir / filename
            
            content = f"# {issue['title']}\n\n"
            content += f"**Labels:** {', '.join(issue['labels'])}\n"
            content += f"**Priority:** {issue['priority']}\n\n"
            content += "---\n\n"
            content += issue['body']
            
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(content)
                
        print(f"📝 {len(self.issues)} Issue-Dateien generiert")
        
    def _generate_summary(self):
        """Generiert Summary-Report"""
        summary_path = self.output_dir / "SUMMARY.md"
        
        # Gruppierung nach Priorität
        by_priority = {}
        for issue in self.issues:
            priority = issue['priority']
            if priority not in by_priority:
                by_priority[priority] = []
            by_priority[priority].append(issue)
            
        content = "# GitHub Issues Summary (Python Generated)\n\n"
        content += f"Generiert: {datetime.now().strftime('%d.%m.%Y, %H:%M:%S')}\n\n"
        content += "## Übersicht\n\n"
        content += f"- **Gesamt Issues:** {len(self.issues)}\n"
        content += f"- **Hohe Priorität:** {len(by_priority.get('high', []))}\n"
        content += f"- **Mittlere Priorität:** {len(by_priority.get('medium', []))}\n"
        content += f"- **Niedrige Priorität:** {len(by_priority.get('low', []))}\n\n"
        
        for priority in ['high', 'medium', 'low']:
            if priority in by_priority:
                content += f"## {priority.title()} Priority Issues\n\n"
                for j, issue in enumerate(by_priority[priority]):
                    filename = f"issue-{self.issues.index(issue)+1:03d}-{self._slugify(issue['title'])}.md"
                    content += f"{j+1}. [{issue['title']}]({filename})\n"
                content += "\n"
                
        content += "## GitHub CLI Commands\n\n"
        content += "Siehe `create-issues.sh` für automatisierte Issue-Erstellung.\n"
        
        with open(summary_path, 'w', encoding='utf-8') as f:
            f.write(content)
            
        print(f"📊 Summary-Report erstellt: {summary_path}")
        
    def _generate_github_cli_script(self):
        """Generiert GitHub CLI Script"""
        script_path = self.output_dir / "create-issues.sh"
        
        script = "#!/bin/bash\n\n"
        script += "# GitHub Issues Auto-Creation Script (Python Generated)\n"
        script += f"# Generiert: {datetime.now().isoformat()}\n\n"
        script += "set -e\n\n"
        script += 'echo "🚀 Erstelle GitHub Issues aus Roadmap..."\n\n'
        
        for i, issue in enumerate(self.issues):
            filename = f"issue-{i+1:03d}-{self._slugify(issue['title'])}.md"
            title_escaped = issue['title'].replace('"', '\\"')
            labels_str = '","'.join(issue['labels'])
            
            script += f'echo "Creating issue {i+1}/{len(self.issues)}: {issue["title"]}"\n'
            script += "gh issue create \\\n"
            script += f'  --title "{title_escaped}" \\\n'
            script += f'  --body-file "{filename}" \\\n'
            script += f'  --label "{labels_str}" \\\n'
            script += f'  --assignee "@me" || echo "Failed to create issue: {issue["title"]}"\n\n'
            
            # Rate limiting
            if (i + 1) % 10 == 0:
                script += 'echo "⏸️  Kurze Pause für Rate Limiting..."\n'
                script += "sleep 5\n\n"
                
        script += 'echo "✅ Issue-Erstellung abgeschlossen!"\n'
        
        with open(script_path, 'w', encoding='utf-8') as f:
            f.write(script)
            
        # Ausführbar machen
        os.chmod(script_path, 0o755)
        
        print(f"🔧 Batch-Script erstellt: {script_path}")
        
    def _slugify(self, text: str) -> str:
        """Erstellt URL-freundlichen Slug"""
        # Deutsche Umlaute ersetzen
        replacements = {'ä': 'ae', 'ö': 'oe', 'ü': 'ue', 'ß': 'ss'}
        for old, new in replacements.items():
            text = text.replace(old, new)
            
        # Nur alphanumerische Zeichen und Bindestriche
        text = re.sub(r'[^a-zA-Z0-9]+', '-', text.lower())
        text = text.strip('-')
        
        return text[:50]  # Maximale Länge


if __name__ == "__main__":
    generator = RoadmapIssueGenerator()
    generator.generate_issues()

File: scripts/generate-issues.js
#!/usr/bin/env node

/**
 * GitHub Issue Generator für NdcApp Roadmap
 * 
 * Dieses Script parst die ROADMAP.md und generiert formatierte GitHub Issues
 * für alle noch nicht erledigten Aufgaben.
 */

const fs = require('fs');
const path = require('path');

// Konfiguration
const ROADMAP_FILE = '../ROADMAP.md';
const OUTPUT_DIR = './generated-issues';
const MAX_ISSUES_PER_BATCH = 20; // GitHub Rate Limiting berücksichtigen

// Issue-Kategorien basierend auf Roadmap-Struktur
const ISSUE_CATEGORIES = {
  'code-quality': {
    label: 'code-quality',
    template: 'code-improvement',
    priority: 'high'
  },
  'documentation': {
    label: 'documentation',
    template: 'documentation',
    priority: 'medium'
  },
  'deployment': {
    label: 'deployment',
    template: 'feature',
    priority: 'high'
  },
  'feature': {
    label: 'enhancement',
    template: 'feature',
    priority: 'medium'
  },
  'ui-ux': {
    label: 'ui/ux',
    template: 'feature',
    priority: 'medium'
  },
  'performance': {
    label: 'performance',
    template: 'code-improvement',
    priority: 'high'
  },
  'technical-debt': {
    label: 'tech-debt',
    template: 'code-improvement',
    priority: 'low'
  }
};

// Prioritäts-Mapping
const PRIORITY_MAPPING = {
  'Kurzzeitige Ziele': 'high',
  'Mittelfristige Ziele': 'medium', 
  'Langfristige Ziele': 'low',
  'Technische Schulden': 'medium'
};

class IssueGenerator {
  constructor() {
    this.roadmapContent = '';
    this.issues = [];
    this.currentSection = '';
    this.currentSubsection = '';
  }

  /**
   * Hauptfunktion zur Generierung aller Issues
   */
  async generateAllIssues() {
    try {
      console.log('🚀 Starte GitHub Issue Generierung...');
      
      this.loadRoadmap();
      this.parseRoadmap();
      this.createOutputDirectory();
      this.generateIssueFiles();
      this.generateSummaryReport();
      this.generateBatchCommands();
      
      console.log(`✅ ${this.issues.length} Issues erfolgreich generiert!`);
      console.log(`📁 Ausgabeverzeichnis: ${OUTPUT_DIR}`);
      
    } catch (error) {
      console.error('❌ Fehler bei der Issue-Generierung:', error.message);
      process.exit(1);
    }
  }

  /**
   * Lädt die Roadmap-Datei
   */
  loadRoadmap() {
    const roadmapPath = path.resolve(__dirname, ROADMAP_FILE);
    
    if (!fs.existsSync(roadmapPath)) {
      throw new Error(`Roadmap-Datei nicht gefunden: ${roadmapPath}`);
    }
    
    this.roadmapContent = fs.readFileSync(roadmapPath, 'utf8');
    console.log('📖 Roadmap-Datei geladen');
  }

  /**
   * Parst die Roadmap und extrahiert alle Tasks
   */
  parseRoadmap() {
    const lines = this.roadmapContent.split('\n');
    let currentSection = '';
    let currentSubsection = '';
    let currentCategory = 'feature';
    let isInTaskList = false;
    
    for (let i = 0; i < lines.length; i++) {
      const line = lines[i].trim();
      
      // Section-Header erkennen (## oder ###)
      if (line.startsWith('## ')) {
        currentSection = line.replace('## ', '').replace(/[🎯🚀🏗️🔧📊🚦📞💡]/g, '').trim();
        currentCategory = this.determineCategoryFromSection(currentSection);
        isInTaskList = this.shouldProcessSection(currentSection);
        continue;
      }
      
      if (line.startsWith('### ')) {
        currentSubsection = line.replace('### ', '').replace(/[0-9.]/g, '').trim();
        continue;
      }
      
      // Tasks erkennen (- [ ] Format)
      if (isInTaskList && line.startsWith('- [ ] ')) {
        const task = this.parseTask(line, lines, i, currentSection, currentSubsection, currentCategory);
        if (task) {
          this.issues.push(task);
        }
      }
    }
    
    console.log(`📋 ${this.issues.length} Tasks aus Roadmap extrahiert`);
  }

  /**
   * Bestimmt die Kategorie basierend auf dem Section-Namen
   */
  determineCategoryFromSection(section) {
    const sectionLower = section.toLowerCase();
    
    if (sectionLower.includes('code') || sectionLower.includes('qualität')) {
      return 'code-quality';
    }
    if (sectionLower.includes('dokumentation') || sectionLower.includes('readme')) {
      return 'documentation';
    }
    if (sectionLower.includes('deployment') || sectionLower.includes('build')) {
      return 'deployment';
    }
    if (sectionLower.includes('ui') || sectionLower.includes('ux') || sectionLower.includes('design')) {
      return 'ui-ux';
    }
    if (sectionLower.includes('performance') || sectionLower.includes('optimierung')) {
      return 'performance';
    }
    if (sectionLower.includes('schulden') || sectionLower.includes('technical debt')) {
      return 'technical-debt';
    }
    
    return 'feature';
  }

  /**
   * Prüft ob eine Section verarbeitet werden soll
   */
  shouldProcessSection(section) {
    const sectionsToProcess = [
      'Kurzzeitige Ziele',
      'Mittelfristige Ziele', 
      'Langfristige Ziele',
      'Technische Schulden',
      'Nächste Schritte'
    ];
    
    return sectionsToProcess.some(s => section.includes(s));
  }

  /**
   * Parst einen einzelnen Task
   */
  parseTask(line, lines, index, section, subsection, category) {
    const taskText = line.replace('- [ ] ', '').replace(/\*\*(.*?)\*\*/g, '$1').trim();
    
    if (!taskText || taskText.length < 10) {
      return null;
    }

    // Sammle zusätzliche Details aus nachfolgenden Zeilen
    const details = this.collectTaskDetails(lines, index + 1);
    
    return {
      title: this.generateIssueTitle(taskText),
      body: this.generateIssueBody(taskText, details, section, subsection),
      labels: this.generateLabels(category, section),
      priority: this.determinePriority(section),
      category: category,
      section: section,
      subsection: subsection,
      rawTask: taskText
    };
  }

  /**
   * Sammelt zusätzliche Details für einen Task
   */
  collectTaskDetails(lines, startIndex) {
    const details = [];
    
    for (let i = startIndex; i < Math.min(startIndex + 5, lines.length); i++) {
      const line = lines[i].trim();
      
      // Stoppe bei nächster Task oder Section
      if (line.startsWith('- [ ]') || line.startsWith('- [x]') || 
          line.startsWith('#') || line === '') {
        break;
      }
      
      // Sammle Sub-Items
      if (line.startsWith('  - ') || line.startsWith('    - ')) {
        details.push(line.replace(/^\s*- /, ''));
      }
    }
    
    return details;
  }

  /**
   * Generiert Issue-Titel
   */
  generateIssueTitle(taskText) {
    // Entferne Formatierung und kürze auf max 80 Zeichen
    let title = taskText.replace(/\*\*(.*?)\*\*/g, '$1');
    
    if (title.length > 80) {
      title = title.substring(0, 77) + '...';
    }
    
    return title;
  }

  /**
   * Generiert Issue-Body
   */
  generateIssueBody(taskText, details, section, subsection) {
    let body = `## Beschreibung\n\n${taskText}\n\n`;
    
    if (subsection) {
      body += `**Bereich:** ${subsection}\n`;
    }
    
    body += `**Roadmap-Kategorie:** ${section}\n\n`;
    
    if (details.length > 0) {
      body += `## Details\n\n`;
      details.forEach(detail => {
        body += `- ${detail}\n`;
      });
      body += '\n';
    }
    
    body += `## Akzeptanzkriterien\n\n`;
    body += `- [ ] Feature/Verbesserung ist implementiert\n`;
    body += `- [ ] Tests sind vorhanden (falls anwendbar)\n`;
    body += `- [ ] Dokumentation ist aktualisiert\n`;
    body += `- [ ] Code Review ist durchgeführt\n\n`;
    
    body += `## Zusätzliche Informationen\n\n`;
    body += `Diese Issue wurde automatisch aus der [Roadmap](../ROADMAP.md) generiert.\n\n`;
    body += `Siehe auch: [Vollständige Roadmap](../ROADMAP.md)`;
    
    return body;
  }

  /**
   * Generiert Labels für Issue
   */
  generateLabels(category, section) {
    const labels = [];
    
    // Basis-Label aus Kategorie
    if (ISSUE_CATEGORIES[category]) {
      labels.push(ISSUE_CATEGORIES[category].label);
    }
    
    // Prioritäts-Label
    const priority = this.determinePriority(section);
    labels.push(`priority-${priority}`);
    
    // Roadmap-Label
    labels.push('roadmap');
    
    return labels;
  }

  /**
   * Bestimmt Priorität basierend auf Section
   */
  determinePriority(section) {
    for (const [key, priority] of Object.entries(PRIORITY_MAPPING)) {
      if (section.includes(key)) {
        return priority;
      }
    }
    return 'medium';
  }

  /**
   * Erstellt Ausgabe-Verzeichnis
   */
  createOutputDirectory() {
    const outputPath = path.resolve(__dirname, OUTPUT_DIR);
    
    if (!fs.existsSync(outputPath)) {
      fs.mkdirSync(outputPath, { recursive: true });
    }
    
    console.log(`📁 Ausgabe-Verzeichnis erstellt: ${outputPath}`);
  }

  /**
   * Generiert Issue-Dateien
   */
  generateIssueFiles() {
    this.issues.forEach((issue, index) => {
      const filename = `issue-${String(index + 1).padStart(3, '0')}-${this.slugify(issue.title)}.md`;
      const filepath = path.join(OUTPUT_DIR, filename);
      
      const content = this.formatIssueFile(issue);
      fs.writeFileSync(filepath, content, 'utf8');
    });
    
    console.log(`📝 ${this.issues.length} Issue-Dateien generiert`);
  }

  /**
   * Formatiert Issue als Markdown-Datei
   */
  formatIssueFile(issue) {
    let content = `# ${issue.title}\n\n`;
    content += `**Labels:** ${issue.labels.join(', ')}\n`;
    content += `**Priority:** ${issue.priority}\n`;
    content += `**Category:** ${issue.category}\n\n`;
    content += `---\n\n`;
    content += issue.body;
    
    return content;
  }

  /**
   * Generiert zusammenfassenden Report
   */
  generateSummaryReport() {
    const summaryPath = path.join(OUTPUT_DIR, 'SUMMARY.md');
    
    let content = `# GitHub Issues Summary\n\n`;
    content += `Generiert: ${new Date().toLocaleString('de-DE')}\n\n`;
    content += `## Übersicht\n\n`;
    content += `- **Gesamt Issues:** ${this.issues.length}\n`;
    
    // Gruppierung nach Priorität
    const byPriority = this.groupBy(this.issues, 'priority');
    content += `- **Hohe Priorität:** ${(byPriority.high || []).length}\n`;
    content += `- **Mittlere Priorität:** ${(byPriority.medium || []).length}\n`;
    content += `- **Niedrige Priorität:** ${(byPriority.low || []).length}\n\n`;
    
    // Gruppierung nach Kategorie
    const byCategory = this.groupBy(this.issues, 'category');
    content += `## Issues nach Kategorie\n\n`;
    
    Object.keys(byCategory).forEach(category => {
      content += `### ${category} (${byCategory[category].length})\n\n`;
      byCategory[category].forEach((issue, index) => {
        const filename = `issue-${String(this.issues.indexOf(issue) + 1).padStart(3, '0')}-${this.slugify(issue.title)}.md`;
        content += `${index + 1}. [${issue.title}](${filename})\n`;
      });
      content += '\n';
    });
    
    content += `## Empfohlene Reihenfolge\n\n`;
    content += `1. **Hohe Priorität zuerst** - Kritische Fixes und Deployment\n`;
    content += `2. **Code-Qualität** - Warnings beheben\n`;
    content += `3. **Dokumentation** - README und Screenshots\n`;
    content += `4. **Features** - Nach Business-Value priorisieren\n\n`;
    
    content += `## GitHub CLI Commands\n\n`;
    content += `Siehe \`create-issues.sh\` für automatisierte Issue-Erstellung.\n`;
    
    fs.writeFileSync(summaryPath, content, 'utf8');
    console.log(`📊 Summary-Report erstellt: ${summaryPath}`);
  }

  /**
   * Generiert Batch-Commands für GitHub CLI
   */
  generateBatchCommands() {
    const scriptPath = path.join(OUTPUT_DIR, 'create-issues.sh');
    
    let script = `#!/bin/bash\n\n`;
    script += `# GitHub Issues Auto-Creation Script\n`;
    script += `# Generiert: ${new Date().toISOString()}\n\n`;
    script += `set -e\n\n`;
    script += `echo "🚀 Erstelle GitHub Issues aus Roadmap..."\n\n`;
    
    // Gruppiere nach Priorität für bessere Reihenfolge
    const priorityOrder = ['high', 'medium', 'low'];
    const byPriority = this.groupBy(this.issues, 'priority');
    
    let issueCount = 0;
    
    priorityOrder.forEach(priority => {
      if (!byPriority[priority]) return;
      
      script += `echo "📋 Erstelle ${priority} Priorität Issues..."\n\n`;
      
      byPriority[priority].forEach((issue, index) => {
        issueCount++;
        const filename = `issue-${String(this.issues.indexOf(issue) + 1).padStart(3, '0')}-${this.slugify(issue.title)}.md`;
        
        script += `echo "Creating issue ${issueCount}/${this.issues.length}: ${issue.title}"\n`;
        script += `gh issue create \\\n`;
        script += `  --title "${issue.title.replace(/"/g, '\\"')}" \\\n`;
        script += `  --body-file "${filename}" \\\n`;
        script += `  --label "${issue.labels.join('","')}" \\\n`;
        script += `  --assignee "@me" || echo "Failed to create issue: ${issue.title}"\n\n`;
        
        // Rate Limiting: Pause nach bestimmter Anzahl
        if (issueCount % 10 === 0) {
          script += `echo "⏸️  Kurze Pause für Rate Limiting..."\n`;
          script += `sleep 5\n\n`;
        }
      });
    });
    
    script += `echo "✅ Issue-Erstellung abgeschlossen!"\n`;
    script += `echo "📊 Überprüfe die erstellten Issues: gh issue list"\n`;
    
    fs.writeFileSync(scriptPath, script, 'utf8');
    fs.chmodSync(scriptPath, 0o755); // Ausführbar machen
    
    console.log(`🔧 Batch-Script erstellt: ${scriptPath}`);
  }

  /**
   * Hilfsfunktion: Gruppiert Array nach Property
   */
  groupBy(array, property) {
    return array.reduce((groups, item) => {
      const key = item[property];
      groups[key] = groups[key] || [];
      groups[key].push(item);
      return groups;
    }, {});
  }

  /**
   * Hilfsfunktion: Erstellt URL-freundlichen Slug
   */
  slugify(text) {
    return text
      .toLowerCase()
      .replace(/[äöüß]/g, match => ({
        'ä': 'ae', 'ö': 'oe', 'ü': 'ue', 'ß': 'ss'
      }[match]))
      .replace(/[^a-z0-9]+/g, '-')
      .replace(/^-+|-+$/g, '')
      .substring(0, 50);
  }
}

// Script ausführen wenn direkt aufgerufen
if (require.main === module) {
  const generator = new IssueGenerator();
  generator.generateAllIssues();
}

module.exports = IssueGenerator;